monai/transforms/spatial/array.py

```diff
@@ -1666,38 +1545,48 @@ def __call__(
             ValueError: When ``grid=None`` and ``spatial_size=None``. Incompatible values.
 
         """
-        if grid is None:  # create grid from spatial_size
-            if spatial_size is None:
-                raise ValueError("Incompatible values: grid=None and spatial_size=None.")
-            grid_ = create_grid(spatial_size, device=self.device, backend="torch", dtype=self.dtype)
+        if not self.lazy_evaluation:
+            if grid is None:  # create grid from spatial_size
+                if spatial_size is None:
+                    raise ValueError("Incompatible values: grid=None and spatial_size=None.")
+                grid_ = create_grid(spatial_size, device=self.device, backend="torch", dtype=self.dtype)
+            else:
+                grid_ = grid
+            _dtype = self.dtype or grid_.dtype
+            grid_: torch.Tensor = convert_to_tensor(grid_, dtype=_dtype, track_meta=get_track_meta())  # type: ignore
+            _device = grid_.device  # type: ignore
+            spatial_dims = len(grid_.shape) - 1
         else:
-            grid_ = grid
-        _dtype = self.dtype or grid_.dtype
-        grid_: torch.Tensor = convert_to_tensor(grid_, dtype=_dtype, track_meta=get_track_meta())  # type: ignore
+            _device = self.device
+            spatial_dims = len(spatial_size)  # type: ignore
         _b = TransformBackends.TORCH
-        _device = grid_.device  # type: ignore
-        affine: NdarrayOrTensor
+        affine: torch.Tensor
         if self.affine is None:
-            spatial_dims = len(grid_.shape) - 1
             affine = torch.eye(spatial_dims + 1, device=_device)
             if self.rotate_params:
-                affine = affine @ create_rotate(spatial_dims, self.rotate_params, device=_device, backend=_b)
+                affine @= create_rotate(spatial_dims, self.rotate_params, device=_device, backend=_b)
             if self.shear_params:
-                affine = affine @ create_shear(spatial_dims, self.shear_params, device=_device, backend=_b)
+                affine @= create_shear(spatial_dims, self.shear_params, device=_device, backend=_b)
             if self.translate_params:
-                affine = affine @ create_translate(spatial_dims, self.translate_params, device=_device, backend=_b)
+                affine @= create_translate(spatial_dims, self.translate_params, device=_device, backend=_b)
             if self.scale_params:
-                affine = affine @ create_scale(spatial_dims, self.scale_params, device=_device, backend=_b)
+                affine @= create_scale(spatial_dims, self.scale_params, device=_device, backend=_b)
         else:
-            affine = self.affine
+            affine = self.affine  # type: ignore
+        affine = to_affine_nd(spatial_dims, affine)
+        if not self.align_corners:
+            affine @= convert_to_dst_type(
```

Comment:
`@=` only works on tensor. For numpy, it will raise the following error:
```
TypeError: In-place matrix multiplication is not (yet) supported. Use 'a = a @ b' instead of 'a @= b'.
```
It seems only this place will meet the issue because "self.affine" could be numpy. I will modify this line only in #6080

Reply:
Hi @yiheng-wang-nv I'm still working on this, if I change the logic again I'll let you know. for now please skip the `align_corner=True` test cases..